templates/flavors/flatcar/patches/kubeadm-controlplane.yaml

```diff
@@ -0,0 +1,63 @@
+---
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+kind: KubeadmControlPlane
+metadata:
+  name: ${CLUSTER_NAME}-control-plane
+spec:
+  kubeadmConfigSpec:
+    diskSetup:
+      filesystems:
+      - device: /dev/disk/azure/scsi1/lun0
+        extraOpts:
+        - -E
+        - lazy_itable_init=1,lazy_journal_init=1
+        filesystem: ext4
+        label: etcd_disk
+        overwrite: false
+      partitions:
+      - device: /dev/disk/azure/scsi1/lun0
+        layout: true
+        overwrite: false
+    format: ignition
+    ignition:
+      containerLinuxConfig:
+        additionalConfig: |
+          systemd:
+            units:
+            - name: kubeadm.service
+              dropins:
+              - name: 10-flatcar.conf
+                contents: |
+                  [Unit]
+                  After=oem-cloudinit.service
+    initConfiguration:
+      nodeRegistration:
+        name: '@@HOSTNAME@@'
+    joinConfiguration:
+      nodeRegistration:
+        name: '@@HOSTNAME@@'
+    mounts:
+    - - etcd_disk
+      - /var/lib/etcddisk
+    postKubeadmCommands: []
+    preKubeadmCommands:
+    - sed -i "s/@@HOSTNAME@@/$(curl -s -H Metadata:true --noproxy '*' 'http://169.254.169.254/metadata/instance?api-version=2020-09-01'
+      | jq -r .compute.name)/g" /etc/kubeadm.yml
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureMachineTemplate
+metadata:
+  name: ${CLUSTER_NAME}-control-plane
+spec:
+  template:
+    spec:
+      image:
+        sharedGallery:
+          gallery: ClusterAPI
+          name: capi-flatcar-stable-2905.2.3
+          offer: flatcar-container-linux-free
+          publisher: kinvolk
+          resourceGroup: cluster-api-images
+          sku: stable
+          subscriptionID: ${AZURE_SUBSCRIPTION_ID}
```

Comment:
@jsturtevant FYI this is a similar issue to what you were encountering in #1729. This template requires an image in a shared gallery since there is currently no published flatcar image in Azure.

@invidian @jepio what are the steps for someone to use this from their own subscription? Do they need to first build an image?

Reply:
>@invidian @jepio what are the steps for someone to use this from their own subscription? Do they need to first build an image?

This is similar to CAPA. Right now we build and publish our own AMIs, but soon they will be build by the CAPA team, as Flatcar's support is included in upstream image-builder.

In general, from what we know about CAPI, users should be responsible for building their own images and images which are already published are intended to be only for testing.

How is this handled for the images for e2e tests which are already used by CAPZ?

Reply:
So far we've been testing manually. This requires building an image in your subscription:
```
cd image-builder/images/capi
make build-azure-sig-flatcar FLATCAR_VERSION=current
```
Then the gallery image parameters need to be set in the template that this PR adds, and you're all set.